substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java

```diff
@@ -141,7 +171,15 @@ public Properties getProperties() {
 
     protected String getProperty(String key) {
         initializeLazyValue(key);
-        return properties.getProperty(key);
+        String ret = properties.getProperty(key);
+        if (ret == null && isNotSupported(key)) {
+            System.out.println("Java system property " + key + " is not available in native image. Please explicitly assign the expected value via -D" + key + "=, or avoid using this property.");
+        }
+        return ret;
+    }
+
+    public static boolean isNotSupported(String key) {
+        return Arrays.stream(UNSUPPORTED_PROPERTIES).anyMatch(s -> s.equals(key));
```

Comment:
General comment: core parts of SVM cannot use streams, because that makes so much of the JDK reachable that the size of a "hello world" explodes.

Reply:
fixed